### DISCUSSION TITLE
Problems with handling single quotes?

### DISCUSSION THREAD

**Discussion by SanokKule** [rejetto/hfs#166] [Q&A] [answered]
```onclick='videoplay("${entry.n}")'``` will not be executed properly, ending on ```videoplay("```?

**Comment by rejetto** [comment]
try this way
`onclick="videoplay(${JSON.stringify(entry.n)})" `

**Comment by SanokKule** [comment]
Now it gets passed empty string i think
`Uncaught SyntaxError: Invalid or unexpected token (at (index):1:11)`
`videoplay()`
Only on files with single quotes in the name.

**Comment by rejetto** [comment, marked as answer]
well, i experimented a bit, and the correct way should be

    `<button onclick='videoplay(${JSON.stringify(entry.n).replace(/'/g, '&apos;')})'>+</button>`

Please confirm it works for you.
It is rather complicated, I know, so I decided that from next version you'll be able to do this

    HFS.onEvent('afterEntryName', ({ entry }, { _ }) =>
        `<button onclick='videoplay("${_.escape(entry.n)}")'>+</button>`)

Anyway, this problem is due to the fact that we are trying to produce correct HTML.
The plugin can generate content as HTML, but also with [pure javascript or React](https://github.com/rejetto/hfs/blob/main/dev-plugins.md#javascript).
In fact, my simple-player plugin has no such problem, as it works with react

    h('button',{ className: 'play-button', onClick: () => play(entry.n) }))

**Reply by SanokKule** [reply]
If i move the html to custom.html with `'videoplay(${JSON.stringify(entry.n).replace(/'/g, '&apos;')})'`  it will still error out on single quotes.
If i add it in the `HFS.onEvent()` it will error out with:
```
media-player.js:20
Uncaught ReferenceError: entry is not defined
    at media-player.js:20:92
    at HTMLDocument.<anonymous> (index-9f5a7cc5.js:93:35655)
    at GR (index-9f5a7cc5.js:93:35403)
    at index-9f5a7cc5.js:59:3376
    at Object.useMemo (index-9f5a7cc5.js:30:25858)
    at de.useMemo (index-9f5a7cc5.js:9:6208)
    at Xi (index-9f5a7cc5.js:59:3355)
    at np (index-9f5a7cc5.js:30:19529)
    at G1 (index-9f5a7cc5.js:32:43986)
    at V1 (index-9f5a7cc5.js:32:39747)

**Reply by SanokKule** [reply]
I will try your way then, and report back

**Reply by SanokKule** [reply]
Yeah, this works great
